src/Registration/RegistrationService.php

```diff
@@ -61,6 +64,8 @@ public function register(RequestInterface $request): ResponseInterface
                 $this->shopSecretGeneratorInterface->generate()
             );
 
+            $this->checkIfShopURLIsReachable($shop);
```

Comment:
I would publish here an event through the eventDispatcher and add an listener in symfony bundle to throw an exception when the shop URL is not reachable

Reply:
I also thought about it, but what if the messenger transport is configured to be async?

Reply:
events are synchronous, events are not messages in symfony :)

Reply:
Right, we have the `EventDispatcherInterface` instead of the `MessageBusInterface` 🤦 

Let me refactor this code 👍